File: src-tauri/tests/fixtures/ast/sample.go
package sample

import "fmt"

func StandaloneFunc(x int) int {
	return x + 1
}

type Counter struct {
	count int
}

func (c *Counter) Increment() {
	c.count++
}

func (c Counter) Value() int {
	return c.count
}

type Adder interface {
	Add(a, b int) int
}

var _ = fmt.Println

File: src-tauri/tests/fixtures/ast/sample.rs
pub fn top_fn(x: u32) -> u32 {
    x + 1
}

pub struct Foo {
    pub x: u32,
}

pub enum Bar {
    A,
    B,
}

pub struct Counter {
    count: u32,
}

impl Counter {
    const MAX: u32 = 999;

    pub fn new() -> Self {
        Counter { count: 0 }
    }

    pub fn increment(&mut self) {
        self.count += 1;
    }

    pub fn value(&self) -> u32 {
        self.count
    }
}
